### PR TITLE
updated README example to match current API

### PR DESCRIPTION
**Pull request by StevenMHernandez**
## Description
The goal of this pull request is to update the README example to match more closely the current API for the library.

NOTE: one thing I found is that I need to keep adding dependencies which I think should be resolved by  `org.openmined.kotlinsyft:syft:0.1.3` itself. For example, I needed to add org.pytorch:pytorch_android to by apps dependencies. (https://github.com/StevenMHernandez/TestKotlinSyft/blob/master/app/build.gradle#L33). I also had to add some other dependencies that you would expect pytorch to do automatically. For example:

```
    implementation "io.reactivex.rxjava2:rxjava:2.2.12"
    implementation "com.squareup.okhttp3:okhttp:4.3.1"
    implementation "org.jetbrains.kotlinx:kotlinx-serialization-runtime:0.14.0"
    implementation "com.jakewharton.retrofit:retrofit2-kotlinx-serialization-converter:0.4.0"
    implementation "com.squareup.retrofit2:retrofit:2.7.1"
    implementation "com.squareup.retrofit2:adapter-rxjava2:2.7.1"
```

## Affected Dependencies
None

## How has this been tested?
I built a new app in Android Studio and applied the current version of the README. This would not run immediately. Next, I added the following change which cleared up IDE errors and allowed me to build the app successfully. (https://github.com/StevenMHernandez/TestKotlinSyft/commit/b0a87a040f22a86bbecddcc689424943268a7e6c)

## Checklist
- [X] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [N/A] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [N/A] My changes are covered by tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,18 +68,16 @@ As a developer, there are few steps to building your own secure federated learni
 You can use KotlinSyft as a front-end or as a background service. The following is a quick start example usage:
 
 ```kotlin
-    val userId = "my Id"
-
     // Optional: Make an http request to your server to get an authentication token
-    val authToken = apiClient.requestToken("https://www.mywebsite.com/request-token/$userId")
```
Comment (StevenMHernandez) [inline README.md:74]: `apiClient.requestToken()` does not appear in this library, so I removed it. Maybe it comes from a dependency?

**Comment by mccorby** [comment]
Hi @StevenMHernandez ,
Thanks for this PR. It surfaced an important issue with the dependencies and we should address it in the library.

I have created an issue to look into this as soon as possible.

#277

File: README.md (hunk 1 above)
Comment (mccorby) [reply README.md:74]: Good catch. This has changed and now it is the responsibility of the client to implement the authentication. The demo app uses a hard-coded value for it.

**Review by mccorby — APPROVED** [review]
LGTM

File: README.md [hunk 2]
```diff
@@ -119,37 +117,39 @@ You can use KotlinSyft as a front-end or as a background service. The following
...
             // Once training finishes generate the model diff
-            val diff = mnistJob.createDiff()
+            val diff = newJob.createDiff()
             // Report the diff to PyGrid and finish the cycle
-            mnistJob.report(diff)
+            newJob.report(diff)
         }
 
-        override fun onRejected() {
+        fun onRejected() {
```
Comment (vkkhare) [inline README.md:152]: onRejected should be available as subscriber method. I think override should be there

**Comment by vkkhare** [comment]
Awesome thanks for pointing it out. I think the issue is because I allowed the developer to specify their own versions of the libs and kotlinsyft to hse them if available. I would need to add a back up dependency there

**Comment by vkkhare** [comment]
I understand the need to shift pytorch to api instead of implementation but rxjava and the rest shouldn't be on us. Rxjava is completely optional for the user and his dependency may have different version than ours. Same with okhttp, retrofit. 

From the looks at your repo I see you only added pytorch and rxjava. The other libs if needed must be totally different from ours. Only pytorch version needs to be compatible. @mccorby?

**Comment by mccorby** [comment]
@vkkhare You are right in that only Pytorch lib should be exposed as it's necessary for the users to create an app.
One thing we can do with respect to the other libraries is to document the versions we are using to avoid incompatibilities.

**Comment by StevenMHernandez** [comment]
@vkkhare sorry, I have some local changes which I did not commit yet. The change involves adding the list of dependencies which I mentioned above. Actually, I think there are even more dependencies which are needed. I stopped manually adding these dependencies because I thought there was a bigger issue going on. 

The thing is, I needed to add RXjava and pytorch for the app to build successfully. However, running the app after the build caused some run time exceptions indicating that syft was unable to access these specific dependencies. 

I will update my TestKotlinSyft repo and post an example of the run time errors.

**Comment by StevenMHernandez** [comment]
> I will update my TestKotlinSyft repo and post an example of the run time errors.

I updated my example repo with some of the additional dependencies. Even more dependencies are needed though.  I just stopped after adding these 5, because it seemed like the wrong way to do it.

If we did not add the above dependencies (i.e. if we used commit https://github.com/StevenMHernandez/TestKotlinSyft/commit/f75c6b7e893e33be1ebcd111dc4fa3e2f4fb66de), then we would get a run-time exception such as:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.stevenmhernandez.testkotlinsyft, PID: 22731
    java.lang.NoClassDefFoundError: Failed resolution of: Lokhttp3/Request$Builder;
        at org.openmined.syft.networking.clients.SyftWebSocket.<init>(SyftWebSocket.kt:36)
        at org.openmined.syft.networking.clients.SyftWebSocket.<init>(SyftWebSocket.kt:27)
        at org.openmined.syft.networking.clients.SocketClient.<init>(SocketClient.kt:49)
        at org.openmined.syft.networking.clients.SocketClient.<init>(SocketClient.kt:42)
        at org.openmined.syft.domain.SyftConfiguration$SyftConfigBuilder.<init>(SyftConfiguration.kt:58)
        at org.openmined.syft.domain.SyftConfiguration$Companion.builder(SyftConfiguration.kt:31)
        at com.stevenmhernandez.testkotlinsyft.MainActivity.onCreate(MainActivity.kt:33)
        at android.app.Activity.performCreate(Activity.java:6977)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1126)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2946)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3064)
        at android.app.ActivityThread.-wrap14(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1659)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6823)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1563)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1451)
```

File: README.md (hunk 2 above)
Comment (StevenMHernandez) [reply README.md:152]: I removed this because the IDE complained about it. I can put it back if you prefer.

Comment (vkkhare) [reply README.md:152]: I see because it's actually onRejected(timeout: String)

Comment (StevenMHernandez) [reply README.md]: Oh! I should have noticed that. Thanks for finding this. I just updated the signature.

**Comment by vkkhare** [comment]
closing this in response to newer API changes.

**Pull request closed without merging**